src/app/sidenav/sidenav.component.ts

```diff
@@ -24,9 +28,40 @@ export class SidenavComponent implements OnInit {
   constructor() { }
 
   ngOnInit() {
+    this.communities = new Array();
+    this.communitiesFavorited = new Array();
+    // populate 20 random communities
+    for (let i = 0; i < 20; i++) {
+      this.communities.push({ communityname: this.communityName = 'Community ' + i, 
+      isFavorited: this.isFavorited, 
+      favouriteCount: Math.floor(Math.random()*(101))});
+    }
+    for (let community of this.communities) {
+      if (community.isFavorited === true) this.communitiesFavorited.push(community);
+    }
+    this.communities.sort((a, b) => b.favouriteCount - a.favouriteCount);
+    this.communitiesFavorited.sort((a, b) => a.communityname.localeCompare(b.communityname));
+    console.log(this.communities);
   }
-
+  
   toggle() {
     this.sidenav.toggle();
   }
-}
+
+  favorite(i: any) {
+    this.communities[i].isFavorited = true;
+    this.communitiesFavorited.push(this.communities[i]);
+    this.communitiesFavorited.sort((a, b) => a.communityname.localeCompare(b.communityname));
+  }
+  removeFavorite(i: any) {
+    this.communities[i].isFavorited = false;
+    let index = this.communitiesFavorited.indexOf(this.communities[i]);
+    if (index !== -1) this.communitiesFavorited.splice(index, 1);
+  }
+  removeFavoriteFromFav(i: any) {
+    this.communitiesFavorited[i].isFavorited = false;
+    let index = this.communities.indexOf(this.communitiesFavorited[i]);
+    if (index !== -1) this.communities.splice(index, 1);
+    this.communitiesFavorited.splice(i, 1);
+  }
```

Comment:
For these functions, I'd recommend passing in the community reference rather than the index. Dealing with indices may become annoying in the future.

Reply:
So for ex. the functions should be:
```
favourite (community) {
...
}
```
rather than 
```
favourite (i) {
...
}
```